### PR TITLE
(DOCSP-9711): Connect to a MongoDB Deployment Using the VSCE

### PR DESCRIPTION
**Pull request by jwilliams-mongo**
[Jira](https://jira.mongodb.org/browse/DOCSP-9711)
[Stage](https://docs-mongodbcom-staging.corp.mongodb.com/1b6a53d/mongodb-vscode/docsworker/DOCSP-9711/connect)

Note to Jeff: I added some includes and replacements that might come in handy as this project continues :)

### CODE REVIEW

File: source/includes/steps-starting-vsce-paste-string.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+level: 4
+ref: vsce-connect-paste
+content: |
+
+  .. important::
+
+     Ensure that you update the connection string with credentials
+     of a database user for your deployment. The connection might 
+     succeed without these credentials, but you won't be able to
+     interact with data in your deployment.
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-paste-string.yaml:52]: This block is confusing to me. What if I'm running without authentication? (shame on me, but lets say I'm just testing locally). Should we reframe this to say "If your deployment requires authentication, ensure that you update the connection string with credentials of a database user for your deployment."

File: source/connect.txt [hunk 2]
```diff
@@ -11,3 +11,129 @@ Connect to Your MongoDB Deployment
...
+.. admonition:: MongoDB Atlas
+   :class: note
+
+   If you need to create a MongoDB host, consider using 
+   `MongoDB Atlas <https://www.mongodb.com/cloud/atlas?tck=docs_vsce>`__.
+   Atlas is a cloud-hosted database-as-a-service which requires no 
+   installation, offers a free tier to get started, and provides a 
+   copyable URI to easily connect |vsce| to your deployment.
+
+If you need to install |vsce|, see :ref:`install` for instructions.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/connect.txt]: fixed

File: source/connect.txt [hunk 3]
```diff
@@ -11,3 +11,129 @@ Connect to Your MongoDB Deployment
...
+
+- When connecting |vsce| to a replica set, it is not
+  recommended to connect directly to an individual replica
+  set member. If the member to which you are connected switches from a
+  :manual:`primary </core/replica-set-primary/>` member to a
+  :manual:`secondary </core/replica-set-secondary/>` or vice versa as
+  the result of an election, |vsce| may either forcibly close
+  the connection or display stale data.
+
+  Instead, to connect to a replica set, use either the replica set
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/connect.txt]: copied this from the compass docs :)

Updated

File: source/connect.txt [hunk 4]
```diff
@@ -11,3 +11,129 @@ Connect to Your MongoDB Deployment
...
+  :guilabel:`SRV record` or :guilabel:`Replica Set Name` when 
+  filling in your connection information.
+
+- You can't connect |vsce| directly to an analytics node.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/connect.txt]: the term directive doesn't seem to work in this repo. The text wrapped in this role didn't render. Used an `:atlas:` link instead to the FAQ entry as I couldn't figure out how to easily link to a glossary term.

File: source/connect.txt [hunk 5]
```diff
@@ -11,3 +11,129 @@ Connect to Your MongoDB Deployment
...
+.. _vsce-connect-task:
+
+Create a Connection to a Deployment
+-----------------------------------
+
+|vsce| provides two methods to create a connection to a deployment. You 
+can: 
+
+- Provide your deployment connection string, or 
+- Fill in your deployment information in specific fields.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/connect.txt]: updated

File: source/connect.txt [hunk 6]
```diff
@@ -11,3 +11,129 @@ Connect to Your MongoDB Deployment
...
+      .. include:: /includes/steps/disconnect-mongodb-view.rst
+
+Disconnecting from a MongoDB instance closes the |vsce| connection to 
+the active instance. You must :ref:`reconnect <vsce-connect-task>` to 
+interact with data in your deployment again.
+
+Remove a Connection
+-------------------
+
+|vsce| provides two methods to remove a connection to a deployment.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/connect.txt]: updated

File: source/includes/steps-remove-connection-mdb-view.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
+source:
+  file: steps-open-mongodb-view.yaml
+  ref: open-mongodb-view
+ref: disconnect-open-mongodb-view
+---
+title: "Right-click the connection you want to remove, then click :guilabel:`Remove Connection."
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-remove-connection-mdb-view.yaml]: updated

File: source/includes/steps-source-command-palette.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
+---
+title: "In |vscode-short|, open the Command Palette."
+level: 4
+ref: vsce-command-palette
+content: |
+
+  .. include:: /includes/list-tables/command-palette-options.rst
+---
+title: "From the Command Palette, chose where you want to save the
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-source-command-palette.yaml]: updated

File: source/includes/steps-source-command-palette.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+     * - Save the connection globally on vscode
+       - The connection is available whenever you open |vscode-short|.
+
+     * - Don't save this connection
+       - The connection is available until you close |vscode-short|.
+
+  .. admonition:: Required Access
+     :class: important
+
+    Once you are connected to your MongoDB deployment, you may
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-source-command-palette.yaml]: updated

File: source/includes/steps-source-command-palette.yaml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    require specific :manual:`user roles <reference/built-in-roles/>`
+    to access various |vsce| features. For more information
+    on the required roles for |vsce| features, see
+    :ref:`required-access`.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-source-command-palette.yaml]: removed. we link to the relevant section in the manual earlier in this note.

File: source/includes/steps-starting-vsce-paste-string.yaml (hunk 1 above)
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-paste-string.yaml]: updated

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 11]
```diff
@@ -0,0 +1,405 @@
...
+title: Create a connection to a deployment.
+level: 4
+ref: create-connection
+content: |
+  If you have no connections, click :guilabel:`Add Connection`.
+
+  If you have at least one connection, but you want to add another one,
+  choose one of the following:
+
+  - Click the :guilabel:`Add MongoDB Connection` icon.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: updated

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 12]
```diff
@@ -0,0 +1,405 @@
...
+  file: steps-open-mongodb-view.yaml
+  ref: open-mongodb-view
+---
+title: Create a connection to a deployment.
+level: 4
+ref: create-connection
+content: |
+  If you have no connections, click :guilabel:`Add Connection`.
+
+  If you have at least one connection, but you want to add another one,
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: updated. went with the case that is applicable all of the time.

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 13]
```diff
@@ -0,0 +1,405 @@
...
+                 authentication mechanism (*New in MongoDB 4.0*). If
+                 selected, you must provide the :guilabel:`Username`,
+                 :guilabel:`Password`, and :manual:`Authentication
+                 Database
+                 </core/security-users/#user-authentication-database>`
+                 to authenticate the user.
+
+                 For more information on the ``SCRAM``
+                 authentication mechanism, see :manual:`SCRAM
+                 </manual/core/security-scram/>`.
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: updated. from compass docs -- same issue might exist there.

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 14]
```diff
@@ -0,0 +1,405 @@
...
+                 authentication mechanism. If selected, you must
+                 provide the  :guilabel:`Username` to authenticate
+                 the user.
+
+                 .. note::
+
+                    If you are using 
+                    :atlas:`Atlas-managed certificates </security-add-mongodb-users/#select-an-authentication-method>`
+                    , your username must be prefaced by "CN="
+                    per :rfc:`2253`. For example, the username
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: updated and added to spreadsheet.

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 15]
```diff
@@ -0,0 +1,405 @@
...
+                    , your username must be prefaced by "CN="
+                    per :rfc:`2253`. For example, the username 
+                    "X509User" must be provided as "CN=X509User".
+
+---
+title: "(Optional) Specify replica set and security connection options."
+level: 4
+ref: connect-additional-options
+content: |
+  The :guilabel:`More Options` connection screen allows you to specify
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: yes. updated.

File: source/includes/steps-starting-vsce-individual-fields.yaml [hunk 16]
```diff
@@ -0,0 +1,405 @@
...
+
+     * - :guilabel:`Replica Set Name`
+       - If your MongoDB deployment is a replica set, specify the
+         replica set name. Not required if you specify your replica
+         set :guilabel:`Hostname` as an :guilabel:`SRV Record`.
+
+     * - :guilabel:`Read Preference`
+       - Specifies how Compass directs read operations. Options are:
+
+         - ``Primary``,
```
[the remark this answers is not in the document]
Comment (jwilliams-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: updated

File: source/includes/steps-starting-vsce-individual-fields.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (jeff-allen-mongo) [reply source/includes/steps-starting-vsce-individual-fields.yaml]: Ack

File: source/includes/steps-source-command-palette.yaml [hunk 17]
```diff
@@ -0,0 +1,40 @@
+---
+title: "In |vscode-short|, open the Command Palette."
+level: 4
+ref: vsce-command-palette
+content: |
+
+  .. include:: /includes/list-tables/command-palette-options.rst
+---
+title: "From the Command Palette, choose where you want to save the
```
Comment (jeff-allen-mongo) [inline source/includes/steps-source-command-palette.yaml:9]: [nit] You can only do this once you actually click `Connect` to submit. Maybe include that as a step before this one

**Comment by jwilliams-mongo** [comment]
Hi @mmarcon can you review this PR for the VSCE doc when you have some time? Thank you!

File: source/includes/steps-source-command-palette.yaml (hunk 17 above)
Comment (mmarcon) [inline source/includes/steps-source-command-palette.yaml:9]: This is not a thing anymore. We found that the UX was a bit confusing so we skip this step now. In the settings of the extension, users can configure what they want the behavior to be.

image

Comment (jeff-allen-mongo) [reply source/includes/steps-source-command-palette.yaml:9]: @mmarcon, I removed the steps from both tabs and called out the setting in the intro to the [Create a Connection](https://docs-mongodbcom-staging.corp.mongodb.com/5d31072/mongodb-vscode/docsworker/DOCSP-9711/connect#create-a-connection-to-a-deployment) section.

**Pull request merged**